lib/chatbot-api/rest-api.ts

```diff
@@ -34,290 +37,341 @@ export class RestApi extends Construct {
+            "cognito-idp:AdminDeleteUser",
+            "cognito-idp:AdminDisableUser",
+            "cognito-idp:AdminEnableUser",
+            "cognito-idp:SignUp",
+            "cognito-idp:ForgotPassword",
+            "cognito-idp:AdminResetUserPassword",
           ],
-          resources: ["*"],
+          resources: [props.userPool.userPoolArn],
+          effect: iam.Effect.ALLOW,
         })
       );
 
-      if (props.config.bedrock?.roleArn) {
+      props.shared.xOriginVerifySecret.grantRead(apiHandler);
+      props.shared.apiKeysSecret.grantRead(apiHandler);
+      props.shared.configParameter.grantRead(apiHandler);
+      props.modelsParameter.grantRead(apiHandler);
+      props.sessionsTable.grantReadWriteData(apiHandler);
+      props.ragEngines?.uploadBucket.grantReadWrite(apiHandler);
+      props.ragEngines?.processingBucket.grantReadWrite(apiHandler);
+
+      if (props.config.bedrock?.enabled) {
         apiHandler.addToRolePolicy(
           new iam.PolicyStatement({
-            actions: ["sts:AssumeRole"],
-            resources: [props.config.bedrock.roleArn],
+            actions: [
+              "bedrock:ListFoundationModels",
+              "bedrock:ListCustomModels",
+              "bedrock:InvokeModel",
+              "bedrock:InvokeModelWithResponseStream",
+            ],
+            resources: ["*"],
           })
         );
+
+        if (props.config.bedrock?.roleArn) {
+          apiHandler.addToRolePolicy(
+            new iam.PolicyStatement({
+              actions: ["sts:AssumeRole"],
+              resources: [props.config.bedrock.roleArn],
+            })
+          );
+        }
       }
     }
 
-    const chatBotApi = new apigateway.RestApi(this, "ChatBotApi", {
-      endpointTypes: [apigateway.EndpointType.REGIONAL],
-      cloudWatchRole: true,
-      defaultCorsPreflightOptions: {
-        allowOrigins: apigateway.Cors.ALL_ORIGINS,
-        allowMethods: apigateway.Cors.ALL_METHODS,
-        allowHeaders: ["Content-Type", "Authorization", "X-Amz-Date"],
-        maxAge: cdk.Duration.minutes(10),
-      },
-      deploy: true,
-      deployOptions: {
-        stageName: "api",
-        loggingLevel: apigateway.MethodLoggingLevel.INFO,
-        tracingEnabled: true,
-        metricsEnabled: true,
-        throttlingRateLimit: 2500,
+    addPermissions(appSyncLambdaResolver);
+
+    const appSyncApi = new appsync.GraphqlApi(this, "graphql-api", {
+      name: "chatbot-graphql-api",
+      schema: appsync.SchemaFile.fromAsset(
+        "lib/chatbot-api/schema/schema.graphql"
+      ),
+      authorizationConfig: {
+        additionalAuthorizationModes: [
+          {
+            authorizationType: appsync.AuthorizationType.IAM,
+          },
+          {
+            authorizationType: appsync.AuthorizationType.USER_POOL,
+            userPoolConfig: {
+              userPool: props.userPool,
+            },
+          },
+        ],
       },
+      xrayEnabled: true,
     });
```

Comment:
This introduces the following deprecation warning on synthesis:
```
[WARNING] aws-cdk-lib.aws_appsync.GraphqlApiProps#schema is deprecated.
  use Definition.schema instead
  This API will be removed in the next major release.
```

Can this be resolved to avoid introducing calls to deprecated code?